code/game/gamemodes/miniantags/space_turkey/space_turkey.dm

```diff
@@ -0,0 +1,140 @@
+
+	maxHealth = 70
+	health = 70
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	obj_damage = 0
+	melee_damage_lower = 4
+	melee_damage_upper = 7
+	see_in_dark = 8
+	attacktext = "pecks"
+	attack_sound = 'sound/weapons/rapierhit.ogg'
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat = 6)
+	gold_core_spawnable = NO_SPAWN
+	universal_speak = 0
+
+	var/playstyle_string = "<b><font size=3 color='red'>It's a productive shift on the station,<br> \
+							and you are a terrible turkey.</font></b><br> \
+							<b>Objective #1:</b> Hatch as many turkeys as possible.<br> \
+							<b>Objective #2:</b> Annoy the crew. Incessantly.<br> \
+							<b>Objective #3:</b> Don't harm anyone unless you're cornered. <i>(Hint: The Flying Tackle doesn't hurt people.)</i>"
+
+	var/egg_type = /obj/item/reagent_containers/food/snacks/egg/turkey
+
+/mob/living/simple_animal/turkey/space_turkey/electrocute_act(shock_damage, obj/source, siemens_coeff = 1, safety = FALSE, override = FALSE, tesla_shock = FALSE, illusion = FALSE, stun = TRUE)
+		..(shock_damage, source, siemens_coeff*0.4, safety, override, tesla_shock, illusion, stun)
+		//Reduces damage turkeys take from shocks. Put in so shocked doors won't kill them in a second.
+
+/mob/living/simple_animal/turkey/space_turkey/proc/ghost_call()
+	notify_ghosts("A space turkey has been created at [get_area(src)]!", enter_link = "<a href=?src=[UID()];ghostjoin=1>(Click to enter)</a>", source = src, action = NOTIFY_ATTACK)
+
+/mob/living/simple_animal/turkey/space_turkey/New(Loc, CallGhosts = TRUE)
+	. = ..()
+
+	pass_flags &= ~PASSMOB
+	add_language("Vox-pidgin", 1)
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/lay_turkey_egg(null))
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/gobble(null))
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/turkey_tackle(null))
+	if(CallGhosts)
+		ghost_call()
+
+/mob/living/simple_animal/turkey/space_turkey/Topic(href, href_list, hsrc)
+	if(href_list["ghostjoin"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			attack_ghost(ghost)
+
+/obj/item/reagent_containers/food/snacks/egg/turkey
+	name = "egg" //Originally had a snarky turkey egg description; removed to hide eggs with other eggs.
+	desc = "An egg!"
+
+#define TURKEY_HATCH_SPEED 30*4 //30 x # of minutes
+
+/obj/item/reagent_containers/food/snacks/egg/turkey/New()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/reagent_containers/food/snacks/egg/turkey/process()
+	amount_grown += 1
+	if(amount_grown >= TURKEY_HATCH_SPEED)
+		notify_ghosts("A space turkey is ready to hatch in [get_area(src)]!", enter_link = "<a href=?src=[UID()];ghostjoin=1>(Click to enter)</a>", source = src, action = NOTIFY_ATTACK)
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/reagent_containers/food/snacks/egg/turkey/Topic(href, href_list, hsrc)
+	if(href_list["ghostjoin"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			attack_ghost(ghost)
```

Comment:
Attack_ghost (Or attack_anything) in general is a function meant to handle something clicking on it. 

Appropriating it for the ghost joining logic is inappropriate.